error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/JUnitFactoryMethodDeclaration.java

```diff
@@ -0,0 +1,249 @@
+            .addAll(getReturnStatementCommentFixes(tree, factoryMethod, state))
+            .build();
+
+    descriptions.forEach(state::reportMatch);
+    return Description.NO_MATCH;
+  }
+
+  private ImmutableList<Description> getFactoryMethodNameFixes(
+      Name methodName,
+      AnnotationTree methodSourceAnnotation,
+      MethodTree factoryMethod,
+      VisitorState state) {
+    String expectedFactoryMethodName = methodName + "TestCases";
+
+    if (HAS_UNMODIFIABLE_SIGNATURE.matches(factoryMethod, state)
+        || factoryMethod.getName().toString().equals(expectedFactoryMethodName)) {
+      return ImmutableList.of();
+    }
+
+    Optional<String> blocker =
+        findMethodRenameBlocker(
+            ASTHelpers.getSymbol(factoryMethod), expectedFactoryMethodName, state);
+    if (blocker.isPresent()) {
+      reportMethodRenameBlocker(
+          factoryMethod, blocker.orElseThrow(), expectedFactoryMethodName, state);
+      return ImmutableList.of();
+    }
+
+    return ImmutableList.of(
+        buildDescription(methodSourceAnnotation)
+            .setMessage(
+                String.format(
+                    "The test cases should be supplied by a method named `%s`",
+                    expectedFactoryMethodName))
+            .addFix(
+                SuggestedFixes.updateAnnotationArgumentValues(
+                        methodSourceAnnotation,
+                        state,
+                        "value",
+                        ImmutableList.of("\"" + expectedFactoryMethodName + "\""))
+                    .build())
+            .build(),
+        buildDescription(factoryMethod)
+            .setMessage(
+                String.format(
+                    "The test cases should be supplied by a method named `%s`",
+                    expectedFactoryMethodName))
+            .addFix(SuggestedFixes.renameMethod(factoryMethod, expectedFactoryMethodName, state))
+            .build());
+  }
+
+  private void reportMethodRenameBlocker(
+      MethodTree tree, String reason, String suggestedName, VisitorState state) {
+    state.reportMatch(
+        buildDescription(tree)
+            .setMessage(
+                String.format(
+                    "The test cases should be supplied by a method named `%s` (but note that %s)",
+                    suggestedName, reason))
+            .build());
+  }
+
+  private ImmutableList<Description> getReturnStatementCommentFixes(
+      MethodTree testMethod, MethodTree factoryMethod, VisitorState state) {
+    ImmutableList<String> parameterNames =
+        testMethod.getParameters().stream()
+            .map(VariableTree::getName)
+            .map(Object::toString)
+            .collect(toImmutableList());
+
+    String expectedComment = parameterNames.stream().collect(joining(", ", "/* { ", " } */"));
+
+    List<? extends StatementTree> statements = factoryMethod.getBody().getStatements();
+
+    Stream<? extends StatementTree> returnStatementsNeedingComment =
```

Comment:
Can there be more than one statement that needs a comment? Don't think so right? If so, it would require an extra test for sure (or I am missing something). Otherwise, we can improve the code a bit I think.